internal/apicodec/codec_v1.go

```diff
@@ -0,0 +1,100 @@
+package apicodec
+
+import (
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+)
+
+type codecV1 struct {
+	mode     Mode
+	memCodec memCodec
```

Comment:
Why do we need to store the mode into `codeV1` since we could dispatch the `memCodec` in the `NewCodecV1`

Reply:
It was used to implement GetMode
Was thinking that maybe some component would like to know codec mode, but it seems that's not necessary. I'll get rid of it.